mlflow/sklearn/__init__.py

```diff
@@ -749,7 +752,7 @@ def fetch_logged_data(run_id):
             stacklevel=2,
         )
 
-    def fit_mlflow(self, clazz, func_name, *args, **kwargs):
+    def fit_mlflow(original, self, *args, **kwargs):
```

Comment:
Seems like we can replace the start_run/end_run logic with a `manage_run` kind of boolean on `safe_patch` for `patch_fit` safely, and improve the error logic.